src/django_components/autodiscover.py

```diff
@@ -0,0 +1,151 @@
+
+    You can change the files searched in each app (Component modules) by setting `components_modules`
+    to a list of modules (relative to the app). E.g. `components_modules=["components", "nested.components"]`
+    will search in each app for `[app_name]/components.py` and `[app_name]/nested/components.py`. Defaults
+    to `["components"]`.
+
+    Autodiscover makes it possible to map the component module paths. This serves
+    as an escape hatch for when you need to use autodiscover in tests.
+    """
+    from django_components.app_settings import app_settings
+
+    # Allow users to specify directly whether to do search for component module,
+    # but default to AUTODISCOVER settings.
+    if skip_component_modules is None:
+        skip_component_modules = not app_settings.AUTODISCOVER
+
+    if components_modules is None:
+        components_modules = ["components"]
+
+    imported_modules: List[str] = []
+
+    if not skip_component_modules:
+        # Autodetect a components.py file (or other files) in each app directory
+        #
+        # TODO1: THIS IS NOT DOCUMENTED!! - Do we want to keep it or not?
+        #        Basically it searches for "<app_name1>/components.py" in each app.
+        #        Altho I like that it could allow people to dynamically configure which components to import, eg.
+        #        ```py
+        #        if xyz:
+        #            import path.to.component
+        #            import path.to.component2
+        #        else:
+        #            import ...
+        #        ```
+        #
+        # TODO: Maybe we could remove this and modify the logic of `Loader.get_dirs`,
+        #       so `<app_name>/components.py` is among the returned files? In which case
+        #       the `components_modules` arg in this function could be made into a setting
+        #       `COMPONENTS.component_modules`, and resolved in `Loader.get_dirs`.
+        #
+        # TODO2: Or maybe we should move `autodiscover_modules` and loading of libraries OUT from autodiscovery?
+        autodiscover_modules(*components_modules)
+
+        # Autodetect a <component>.py file in components dirs
+        dirs = get_dirs()
+        component_filepaths = search_dirs(dirs, "**/*.py")
+        logger.debug(f"Autodiscover found {len(component_filepaths)} files in component directories.")
+
+        for path in component_filepaths:
+            module_name = _filepath_to_python_module(path)
+            if map_components_modules:
+                module_name = map_components_modules(module_name)
+
+            # This imports the file and runs it's code. So if the file defines any
+            # django components, they will be registered.
+            logger.debug(f'Importing module "{module_name}" (derived from path "{path}")')
+            importlib.import_module(module_name)
+            imported_modules.append(module_name)
+
+    if not skip_libraries:
```

Comment:
It doesn't show up in this diff because I moved the file, but in this WIP I was also playing with the idea of allowing to skip some parts of the `autodiscover`. The reason being that `autodiscover` searches in 3 distinct places. But I prefer more a different approach, the one I've described in https://github.com/EmilStenstrom/django-components/pull/533/files#r1649626042